components/BottomNavigation/tests/unit/beta/MDCBottomNavigationBarControllerTests.m

```diff
@@ -0,0 +1,348 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MDCBottomNavigationBarController.h"
+
+static CGFloat const kDefaultExpectationTimeout = 15;
+
+@interface MDCBottomNavigationControllerTests
+    : XCTestCase <MDCBottomNavigationBarControllerDelegate>
+
+/** The bottom navigation controller to test **/
+@property(nonatomic, strong, nonnull)
+    MDCBottomNavigationBarController *bottomNavigationBarController;
+
+/**
+ * Is fufilled when the a delegate method is called. Set the description of the expectation with the
+ * string value of the selector of the method you are expecting to be called.
+ **/
+@property(nonatomic, strong, nullable) XCTestExpectation *delegateExpecation;
+
+/**
+ * An array of the values of the expected arguments the delegate method should be called with.
+ * The order of the array is the order the arguments appear in the method signature.
+ */
+@property(nonatomic, strong, nullable) NSArray<id> *expectedArguments;
+
+/**
+ * The return value a delegate method should return with.
+ */
+@property(nonatomic, strong, nullable) id delegateReturnValue;
+
+@end
+
+@implementation MDCBottomNavigationControllerTests
+
+- (void)setUp {
+  [super setUp];
+  self.bottomNavigationBarController = [[MDCBottomNavigationBarController alloc] init];
+}
+
+- (void)tearDown {
+  [super tearDown];
```

Comment:
Nit: Please call `[super tearDown]` at the end of your `tearDown` method. It's a safer pattern in case the superclass destroys something you depend on.